examples/bri-3/src/bri/communication/agents/messaging.agent.ts

```diff
@@ -72,12 +89,28 @@ export class MessagingAgent implements OnApplicationBootstrap {
     let newBpiMessageCandidate: BpiMessage;
 
     try {
-      newBpiMessageCandidate = this.parseJsonOrThrow(rawMessage);
+      const newBpiMessageProps = this.parseJsonOrThrow(rawMessage);
+      newBpiMessageCandidate = new BpiMessage(
+        newBpiMessageProps.id,
+        newBpiMessageProps.fromBpiSubjectId,
+        newBpiMessageProps.toBpiSubjectId,
+        newBpiMessageProps.content,
+        newBpiMessageProps.signature,
+        newBpiMessageProps.type,
+      );
     } catch (e) {
       errors.push(`${rawMessage} is not valid JSON. Error: ${e}`);
       return [newBpiMessageCandidate, errors];
     }
 
+    if (
+      !newBpiMessageCandidate.isInfoMessage() &&
+      !newBpiMessageCandidate.isTransactionMessage()
+    ) {
+      errors.push(`type: ${newBpiMessageCandidate.type} is unknown`);
+      return [null, errors];
```

Comment:
is there a reason to not just throw err or antything else, this seems a bit odd for ts

Reply:
We want to collect all errors in one place and want to avoid throw as we are in a background task that should continue to execute